distribution/lib/Standard/Base/0.0.0-dev/src/Network/Email/Email_Provider.enso

```diff
@@ -0,0 +1,53 @@
+import project.Data.Text.Text
+import project.Network.Email.Email_Address.Email_Address
+import project.Network.Email.Email_Body.Email_Body
+import project.Network.Email.Email_Object.Email_Object
+import project.Data.Json.JS_Object
+import project.Network.HTTP.Header.Header
+import project.Data
+import project.Data.Pair.pair
+import project.Network.Email.Email_Send_Response.Send_Response
+import project.System.File.File
+import project.Nothing.Nothing
+
+polyglot java import java.util.Base64
+
+type Email_Provider
+    TestOnly
+    SendGrid api_key:Text
+
+    send self email:Email_Object = case self of
+        Email_Provider.SendGrid api_key -> sendgrid_email email api_key
+        Email_Provider.TestOnly -> Send_Response.Test email
+
+sendgrid_email email:Email_Object api_key:Text =
```

Comment:
- This is implementation, not even [SPI](https://wiki.apidesign.org/wiki/SPI)!
- Hide into own `private` implementation module